src/env-inl.h

```diff
@@ -94,7 +94,7 @@ v8::Local<v8::Array> AsyncHooks::js_execution_async_resources() {
 
 v8::Local<v8::Object> AsyncHooks::native_execution_async_resource(size_t i) {
   if (i >= native_execution_async_resources_.size()) return {};
-  return PersistentToLocal::Strong(native_execution_async_resources_[i]);
+  return native_execution_async_resources_[i];
```

Comment:
Shouldn't we assert that this the local is not empty before using it because we aren't taking ownership of it and it may get invalidated once the JS stack changes?

Reply:
I’m also not sure how pointer invalidation and the local being empty/not being empty are related. Can you clarify that for me?

(And to be clear, an empty `v8::Local<>` is currently a valid return value of this function.)

Reply:
I assumed that the pointers are nullified when they get GCed but I'm not really sure. Are they left untouched? If it is valid to return an empty local then I'm okay with it.

Reply:
> I assumed that the pointers are nullified when they get GCed but I'm not really sure

I don’t think so, no, at least unless V8 is built in debug mode or with `VERIFY_HEAP` enabled.

Reply:
Do you know of a way in which we can make use of https://github.com/nodejs/node/blob/master/deps/v8/src/diagnostics/objects-debug.cc to apply these checks? The `ifdef VERIFY_HEAP` guards enclosed calls to functions from this file, so I was wondering.

Reply:
None of these things are exposed through the API, I think :confused: But really, I think that’s okay here.